crates/hyperswitch_domain_models/src/payments/payment_intent.rs

```diff
@@ -287,39 +293,44 @@ pub enum PaymentIntentUpdate {
         status: storage_enums::IntentStatus,
         updated_by: String,
     },
-}
+    /// UpdateIntent
+    UpdateIntent {
+        amount: Option<MinorUnit>,
+        currency: Option<storage_enums::Currency>,
+        shipping_cost: Option<MinorUnit>,
+        // TODO: Check how to handle this
+        // tax_details: Option<diesel_models::TaxDetails>,
+        skip_external_tax_calculation: Option<super::TaxCalculationOverride>,
+        skip_surcharge_calculation: Option<super::SurchargeCalculationOverride>,
+        surcharge_amount: Option<MinorUnit>,
+        tax_on_surcharge: Option<MinorUnit>,
+        routing_algorithm_id: Option<id_type::RoutingId>,
+        capture_method: Option<common_enums::CaptureMethod>,
+        authentication_type: Option<common_enums::AuthenticationType>,
+        billing_address: Box<Option<Encryptable<Secret<Address>>>>,
+        shipping_address: Box<Option<Encryptable<Secret<Address>>>>,
+        customer_present: Option<common_enums::PresenceOfCustomerDuringPayment>,
+        description: Box<Option<common_utils::types::Description>>,
+        return_url: Box<Option<common_utils::types::Url>>,
+        setup_future_usage: Option<storage_enums::FutureUsage>,
+        apply_mit_exemption: Option<common_enums::MitExemptionRequest>,
+        statement_descriptor: Box<Option<StatementDescriptor>>,
+        order_details: Box<Option<Vec<Secret<diesel_models::types::OrderDetailsWithAmount>>>>,
```

Comment:
Why do we need to make this as a secret? If there is a typed struct, it is the structs responsibility to mask / unmask details.

Reply:
I used `create_intent` as a reference. It was a Secret in `PaymentIntent`, which is why I did it like this.

Reply:
We do not need the field to be a secret if it is typed

Reply:
Since this change would need to be made for `PaymentIntent` struct as well, I think it'd be better to take it up in a separate PR. This one has already been open for too long. Thoughts?